src/utilities/EventEmitter.ts

```diff
@@ -190,10 +190,10 @@ export default class EventEmitter<T extends EventMap> {
    */
   listeners<K extends EventKey<T>>(event: K): Function[] | null {
     if (event) {
-      const listeners = this.events[event] || [];
+      const listeners = [...(this.events[event] ?? [])];
 
       if (isArray(this.eventsOnce[event])) {
-        Array.prototype.push.apply(listeners, this.eventsOnce[event]);
+        listeners.push(...this.eventsOnce[event]);
```

Comment:
These two ways of calling push are not equivalent, I think we should leave the original one. See https://ably-real-time.slack.com/archives/CF7GGBF2N/p1679678739061199

Reply:
Ah. That's unfortunate, although the exception seems to be for non-array elements being pushed. I didn't think this would still be considered idiomatic, although I would expect developers to know what it does. Will change this back.

Reply:
we could change it, but then we should change across the file, not a single place. I haven't really checked perf implications either, so I think it's sensible to leave it in for now.